pkg/networking/backend_sg_provider.go

```diff
@@ -0,0 +1,245 @@
+package networking
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	ec2sdk "github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	networking "k8s.io/api/networking/v1beta1"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/aws/services"
+	"sigs.k8s.io/aws-load-balancer-controller/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	resourceTypeSecurityGroup = "security-group"
+	tagKeyK8sCluster          = "elbv2.k8s.aws/cluster"
+	tagKeySGType              = "elbv2.k8s.aws/type"
```

Comment:
shall we use the `elbv2.k8s.aws/resource` (came from our resourceID of `ingress.k8s.aws/resource` and `service.k8s.aws/resource`

so maybe in the future, when we support multiple controllers in a cluster, the SG can be tagged like
```
elbv2.k8s.aws/cluster: cluster-name
elbv2.k8s.aws/stack: controller-deployment-specific-id
elbv2.k8s.aws/resource: controller-specific-resource-id(in our case , backend-sg)
```

Reply:
using the following tags for now

```
elbv2.k8s.aws/cluster: cluster-name
elbv2.k8s.aws/resource: controller-specific-resource-id(in our case , backend-sg)
```

Will add the stack tag once we add support for multi controller setups.